mesonbuild/interpreter/type_checking.py

```diff
@@ -573,7 +573,8 @@ def _objects_validator(vals: T.List[ObjectTypes]) -> T.Optional[str]:
     OVERRIDE_OPTIONS_KW,
     KwargInfo('build_by_default', bool, default=True, since='0.38.0'),
     KwargInfo('extra_files', ContainerTypeInfo(list, (str, File)), default=[], listify=True),
-    INSTALL_KW,
+    # Accursed. We allow this for backwards compat and warn in the interpreter.
+    KwargInfo('install', object, default=False),
```

Comment:
I think `INSTALL_KW = KwargInfo('install', bool, default=False)` is generally wrong. We always allowed that kwarg to be a list of boolean for the case it has multiple outputs. That's especially useful for CustomTarget, but also for any BuildTarget that could have extra vala outputs IIRC.

Reply:
Oh wait, maybe that was only for install_dir?

Reply:
Right, it is install_dir that allows a list, maybe the warning should hint at doing that, because it very likely was the intention when doing `install: [true, false, get_option('foobar')]`.

I'm wondering why this suddenly happens, the PR that changed this is 2 years old.

Reply:
The PR that what'd this? We do no form of type checking on this kwarg at all. That's why I've written `object` here. It's entirely possible for people to have been using this kwarg with `install: executable('foo', 'foo.c')` which is total nonsense but also a valid `object`.

One of the reports I got was for someone confusing it for install_dir (despite the fact that they use install_dir correctly below). The other report I got was someone doing `install: 'true'` which makes me wonder if people are also doing `install' 'false'` and getting... `bool('false') == True`. Perhaps they were more used to JavaScript....